opentelemetry-sdk/src/metrics/internal/sum.rs

```diff
@@ -43,22 +85,33 @@ impl<T: Number<T>> ValueMap<T> {
             self.no_attribute_value.add(measurement);
             self.has_no_value_attribute_value
                 .store(true, Ordering::Release);
-        } else if let Ok(mut values) = self.values.lock() {
-            let size = values.len();
-            match values.entry(attrs) {
-                Entry::Occupied(mut occupied_entry) => {
-                    let sum = occupied_entry.get_mut();
-                    *sum += measurement;
-                }
-                Entry::Vacant(vacant_entry) => {
-                    if is_under_cardinality_limit(size) {
-                        vacant_entry.insert(measurement);
-                    } else {
-                        values
-                            .entry(STREAM_OVERFLOW_ATTRIBUTE_SET.clone())
-                            .and_modify(|val| *val += measurement)
-                            .or_insert(measurement);
-                        global::handle_error(MetricsError::Other("Warning: Maximum data points for metric stream exceeded. Entry added to overflow.".into()));
+        } else {
+            let bucket_index = Self::hash_to_bucket(&attrs) as usize; // Ensure index is usize for array indexing
+            let bucket_mutex = &self.buckets[bucket_index];
+            let mut bucket_guard = bucket_mutex.lock().unwrap();
+
+            if bucket_guard.is_none() {
+                *bucket_guard = Some(HashMap::new()); // Initialize the bucket if it's None
+            }
+
+            if let Some(ref mut values) = *bucket_guard {
+                let size = values.len();
+                match values.entry(attrs) {
+                    Entry::Occupied(mut occupied_entry) => {
+                        let sum = occupied_entry.get_mut();
+                        *sum += measurement;
+                    }
+                    Entry::Vacant(vacant_entry) => {
+                        if is_under_cardinality_limit(size) {
+                            vacant_entry.insert(measurement);
+                        } else {
+                            // TBD - Update total_count ??
+                            values
+                                .entry(STREAM_OVERFLOW_ATTRIBUTE_SET.clone())
```

Comment:
do we need to store the overflow attribute in each bucket? or once only?

Reply:
overflow attribute is handled separately, and is stored only once.

Reply:
i am not sure how is this handled.. it looks to me like we'll store overflow for each of the hasmaps.

Reply:
Good point, this was indeed overflow every hashmap. Have fixed it now. Will also add tests to validate this.